tests/test_D1_model_components.py

```diff
@@ -718,6 +730,145 @@ def test_storage_fix(self):
+        )
+
+    def test_storage_optimize_investment_minimum_0_float(self):
+        # Test if minimum value is zero if thermal losses are zero (float)
+        dict_asset = self.dict_values[ENERGY_STORAGE]["storage_optimize"]
+        dict_asset[STORAGE_CAPACITY][MAXIMUM_CAP] = {VALUE: None, UNIT: "kWh"}
+        dict_asset[INPUT_POWER][MAXIMUM_CAP] = {VALUE: None, UNIT: "kWh"}
+        dict_asset[OUTPUT_POWER][MAXIMUM_CAP] = {VALUE: None, UNIT: "kWh"}
+        dict_asset[STORAGE_CAPACITY][THERM_LOSSES_REL] = {VALUE: 0, UNIT: "no_unit"}
+        dict_asset[STORAGE_CAPACITY][THERM_LOSSES_ABS] = {VALUE: 0, UNIT: "kWh"}
+
+        D1.storage(
+            model=self.model,
+            dict_asset=dict_asset,
+            bus=self.busses,
+            storage=self.storages,
+        )
+
+        assert (
+            self.model.entities[-1].investment.minimum == 0
+        ), f"investment.minimum should be zero with {THERM_LOSSES_REL} and {THERM_LOSSES_ABS} that equal to zero"
+
+    def test_storage_optimize_investment_minimum_0_time_series(self):
+        # Test if minimum value is zero if thermal losses are zero (time series)
+        dict_asset = self.dict_values[ENERGY_STORAGE]["storage_optimize"]
+        dict_asset[STORAGE_CAPACITY][MAXIMUM_CAP] = {VALUE: None, UNIT: "kWh"}
+        dict_asset[INPUT_POWER][MAXIMUM_CAP] = {VALUE: None, UNIT: "kWh"}
+        dict_asset[OUTPUT_POWER][MAXIMUM_CAP] = {VALUE: None, UNIT: "kWh"}
+        dict_asset[STORAGE_CAPACITY][THERM_LOSSES_REL] = {
+            VALUE: [0, 0, 0, 0],
+            UNIT: "no_unit",
+        }
+        dict_asset[STORAGE_CAPACITY][THERM_LOSSES_ABS] = {
+            VALUE: [0, 0, 0, 0],
+            UNIT: "kWh",
+        }
+
+        D1.storage(
+            model=self.model,
+            dict_asset=dict_asset,
+            bus=self.busses,
+            storage=self.storages,
+        )
+
+        assert (
+            self.model.entities[-1].investment.minimum == 0
+        ), f"investment.minimum should be zero with {THERM_LOSSES_REL} and {THERM_LOSSES_ABS} that equal to zero"
+
+    def test_storage_optimize_investment_minimum_1_rel_float(self):
+        # Test if minimum value is one if relative thermal losses are not zero (float)
+        dict_asset = self.dict_values[ENERGY_STORAGE]["storage_optimize"]
+        dict_asset[STORAGE_CAPACITY][MAXIMUM_CAP] = {VALUE: None, UNIT: "kWh"}
+        dict_asset[INPUT_POWER][MAXIMUM_CAP] = {VALUE: None, UNIT: "kWh"}
+        dict_asset[OUTPUT_POWER][MAXIMUM_CAP] = {VALUE: None, UNIT: "kWh"}
+        dict_asset[STORAGE_CAPACITY][THERM_LOSSES_REL] = {VALUE: 0.001, UNIT: "no_unit"}
+        dict_asset[STORAGE_CAPACITY][THERM_LOSSES_ABS] = {VALUE: 0, UNIT: "kWh"}
+
+        D1.storage(
+            model=self.model,
+            dict_asset=dict_asset,
+            bus=self.busses,
+            storage=self.storages,
+        )
+
+        assert self.model.entities[-1].investment.minimum == 1
```

Comment:
Could you add an assertion message here, too?

Reply:
Oh right! Thank you. :)
Done with commit https://github.com/rl-institut/multi-vector-simulator/pull/718/commits/6005fc7a88893ea15c0bccb70d8c5bd596c8fe5c